osu.Game.Rulesets.Osu/Mods/OsuModZoomed.cs

```diff
@@ -0,0 +1,126 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System;
+using System.Diagnostics;
+using osu.Framework.Bindables;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Sprites;
+using osu.Framework.Utils;
+using osu.Game.Configuration;
+using osu.Game.Rulesets.Mods;
+using osu.Game.Rulesets.Osu.Objects;
+using osu.Game.Rulesets.Scoring;
+using osu.Game.Rulesets.UI;
+using osu.Game.Scoring;
+using osuTK;
+
+namespace osu.Game.Rulesets.Osu.Mods
+{
+    internal class OsuModZoomed : Mod, IUpdatableByPlayfield, IApplicableToScoreProcessor, IApplicableToDrawableRuleset<OsuHitObject>
+    {
+        public override string Name => "Zoomed";
+        public override string Acronym => "ZM";
+        public override IconUsage? Icon => FontAwesome.Solid.Glasses;
+        public override ModType Type => ModType.Fun;
+        public override string Description => "Big brother is watching your cursor.";
+        public override double ScoreMultiplier => 1;
+
+        private const int last_zoom_combo = 200;
+        private const int zoom_every_combo_amount = 100;
+        private const int apply_zoom_duration = 1000;
+        private const double default_zoom = 1.8;
+        private const double zoom_with_combo_by = 0.1;
+
+        [SettingSource("Delay", "Delay in milliseconds for the view to catch up to the cursor")]
+        public BindableInt MovementDelay { get; } = new BindableInt
+        {
+            MinValue = 0,
+            MaxValue = 1000,
+            Precision = 100,
+        };
+
+        [SettingSource("Initial zoom", "The starting zoom level")]
+        public BindableDouble InitialZoom { get; } = new BindableDouble(default_zoom)
+        {
+            MinValue = 1.5,
+            MaxValue = 2,
+            Precision = 0.05
+        };
+
+        [SettingSource("Increase zoom with combo", "Whether zoom should increase as combo increases")]
+        public BindableBool ComboBasedZoom { get; } = new BindableBool(true);
+
+        private double currentZoom;
+
+        private readonly BindableInt currentCombo = new BindableInt();
+
+        private IFrameStableClock? gameplayClock;
+
+        public void ApplyToDrawableRuleset(DrawableRuleset<OsuHitObject> drawableRuleset)
+        {
+            currentZoom = InitialZoom.Value;
```

Comment:
I don't think this is required to be here?

Reply:
Oh, yes it is indeed required, because at the start current zoom isn't being set and it differs from the base zoom which is just the base.